src/server.go

```diff
@@ -76,9 +77,38 @@ func (a *App) QueryHandler(key []byte, w http.ResponseWriter, r *http.Request) {
   }
 }
 
+func (a *App) GetAuthorization(w http.ResponseWriter, r *http.Request) (string, bool) {
+  var authstring string
+  if a.basicauth == true {
+    authtoken := r.Header.Get("Authorization")
+    if authtoken != "" {
+      authstringb, _ := base64.StdEncoding.DecodeString(strings.Split(authtoken, "Basic ")[1])
+      authstring = string(authstringb)
+      authstring = authstring+"@"
+      remote := fmt.Sprintf("http://%s%s", authstring, a.volumes[rand.Intn(len(a.volumes))])
+      resp, err := http.Get(remote)
+      if err != nil {
+        w.WriteHeader(500)
+        return authstring, false
+      }
+      if resp.StatusCode != 200 {
+        w.WriteHeader(resp.StatusCode)
+        return authstring, false
+      }
+    }
+  }
+  return authstring, true
+}
```

Comment:
Nginx performs the basic authorization check before allowing the master to proceed with its operation. It chooses a random volume to perform the verification